Doc/c-api/complex.rst

```diff
@@ -127,12 +127,12 @@ Complex Numbers as Python Objects
 
    Return the :c:type:`Py_complex` value of the complex number *op*.
 
-   If *op* is not a Python complex number object but has a :meth:`__complex__`
+   If *op* is not a Python complex number object but has a :meth:`~object.__complex__`
    method, this method will first be called to convert *op* to a Python complex
-   number object.  If ``__complex__()`` is not defined then it falls back to
-   :meth:`__float__`.  If ``__float__()`` is not defined then it falls back
-   to :meth:`__index__`.  Upon failure, this method returns ``-1.0`` as a real
+   number object.  If :meth:`~object.__complex__` is not defined then it falls back to
+   :meth:`~object.__float__`.  If :meth:`~object.__float__` is not defined then it falls back
```

Comment:
I think it's unnecessary to link to the same think more than once in the same paragraph, and having too many links in one sentence can be distracting. Let's suppress the link (but keep the formatting) for the second occurrence of the `__complex__` and `__float__` methods:

```suggestion
   number object.  If :meth:`!__complex__` is not defined then it falls back to
   :meth:`~object.__float__`.  If :meth:`!__float__` is not defined then it falls back
```